pkglistgen/update_repo_handler.py

```diff
@@ -74,7 +74,13 @@ def parse_repomd(repo, baseurl):
     root = ET.fromstring(repomd.content)
     primary_element = root.find('.//r:data[@type="primary"]', ns)
     location = primary_element.find('r:location', ns).get('href')
-    sha256_expected = primary_element.find('r:checksum[@type="sha256"]', ns).text
+    sha256_or_512 = 0
+    try:
+        sha_expected = primary_element.find('r:checksum[@type="sha256"]', ns).text
```

Comment:
Shouldn't it prefer sha512?

Reply:
Can switch it, for sure. repo data contains either/or

There are currently more files with sha256 being scanned (all released products) vs sha512 (TW).. so more will go the except route...